### PR TITLE
Add nvti_add_tag() for adding a single tag to tags.

### PR DESCRIPTION
**Pull request by janowagner**
So far it was only possible to set the entire tags
string. Now it is possible to append a single tag.

### CODE REVIEW

File: base/nvti.c [hunk 1]
```diff
@@ -1057,6 +1057,46 @@ nvti_set_solution_type (nvti_t *n, const gchar *solution_type)
   return (0);
 }
 
+/**
+ * @brief Add a tag to the NVT tags.
+ *
+ * @param n The NVT Info structure.
+ *
+ * @param name  The tag name. A copy will be created from this.
+ * @param value The tag value. A copy will be created from this.
```
Comment (mattmundell) [inline base/nvti.c]:
```suggestion

 * @param value The tag value. A copy will be created from this.
```

File: base/nvti.c [hunk 2]
```diff
@@ -1057,6 +1057,46 @@ nvti_set_solution_type (nvti_t *n, const gchar *solution_type)
...
+
+  if (n->tag)
+    {
+      new = g_strconcat (n->tag, "|", name, "=", value, NULL);
+      g_free (n->tag);
+    }
+  else
+    new = g_strconcat (name, "=", value, NULL);
+
+  if (new && new[0])
```
Comment (mattmundell) [inline base/nvti.c]: This seems wrong, if you don't have other changes in mind.

new will always have an actual value due to the checks and the two g_strconcat calls above.

I think new should be declared in the if branch and the else should just assign to n->tag.

File: base/nvti.c [hunk 3]
```diff
@@ -1057,6 +1057,46 @@ nvti_set_solution_type (nvti_t *n, const gchar *solution_type)
...
+ *
+ * @param name  The tag name. A copy will be created from this.
+ * @param value The tag value. A copy will be created from this.
+ *
+ * @return 0 for success. Anything else indicates an error.
+ */
+int
+nvti_add_tag (nvti_t *n, const gchar *name, const gchar *value)
+{
+  char *newtag;
```
Comment (mattmundell) [inline base/nvti.c]:
```suggestion
  gchar *newtag;
```

gchar because that's what g_strconcat returns.

Comment (janowagner) [reply base/nvti.c]: Done.

File: base/nvti.c (hunk 2 above)
Comment (janowagner) [reply base/nvti.c]: I condensed this now.

File: base/nvti.c (hunk 1 above)
Comment (janowagner) [reply base/nvti.c]: Hm, I rather preferred to remove the extra empty line to make it consistent.

File: base/nvti.c (hunk 2 above)
Comment (mattmundell) [reply base/nvti.c]: Better, thanks.  newtag is only used in the 'if' branch, so you could
declare it in the 'if' branch, to make that clear.

File: base/nvti.c (hunk 1 above)
Comment (mattmundell) [reply base/nvti.c]: Consistent with what?  :)

The rest of the file uses spacing (except for two functions right at the end).

The rest of gvm-libs does not use the spacing though.

File: base/nvti.c (hunk 2 above)
Comment (janowagner) [reply base/nvti.c]: Done.

File: base/nvti.c (hunk 1 above)
Comment (mattmundell) [reply base/nvti.c]: Will keep in mind for new PR.

**Pull request merged**